### PR TITLE
ALL_USERS_GROUP code cleanup

### PR DESCRIPTION
**Pull request by jaragunde**
I started cleaning up uses of the ALL_USERS_GROUP config parameter, replacing them with the newly added GetAllActiveUsers action / facade operation, and ended up removing some dead code and doing some simplifications in the extra hours code.

### CODE REVIEW

**Comment by jaragunde** [comment]
I tested the extra hours changes with production data, doing a diff of the CSV output of the accumulated extra hours report, before and after the changes.

File: model/facade/action/ExtraHoursReportAction.php [hunk 1]
```diff
@@ -153,145 +152,125 @@ private function numWorkDays(DateTime $init, DateTime $end) {
...
         $userWork = array();
 
-        foreach($users as $userVO)
-        {
-            if (!is_null($userVO->getId()))
-            {
-                $hoursWorked = $taskDao->
-                        getTaskReport($userVO, $init, $end, "USER");
+        $hoursWorked = $taskDao->
+                getTaskReport($userVO, $init, $end, "USER");
```
Comment (jaragunde) [inline model/facade/action/ExtraHoursReportAction.php:169]: Note for reviewer: the chunk of changes starting here is just re-indentation, no actual code was modified.

File: model/facade/action/ExtraHoursReportAction.php [hunk 2]
```diff
@@ -153,145 +152,125 @@ private function numWorkDays(DateTime $init, DateTime $end) {
...
+
+            $work = $this->numWorkDays($row["init"], $row["end"]);
+            $holidays = count($commonDao->
+                    getByCityIdDates($row["city"], $row["init"], $row["end"]));
+            $workHours = ($work - $holidays) * $row["journey"];
+            $userWork["workable_hours"] += $workHours;
         }
 
+        $userWork["extra_hours"] = $userWork["total_hours"] - $userWork["workable_hours"];
+
```
Comment (jaragunde) [inline model/facade/action/ExtraHoursReportAction.php:273]: Note for reviewer: re-indentation ends here.

File: jobs/sendEmailAlertNoFill.php [hunk 3]
```diff
@@ -86,36 +85,32 @@ function sendEmail($email, $lastTaskDate, $from, $cc, $subject, $template) {
                 // User is not currently hired, skip it.
                 continue;
             }
-            $groups = $user->getGroups();
-            foreach ($groups as $group) {
-                if ($group->getName() == $ALL_USERS_GROUP) {
```
Comment (anarute) [inline jobs/sendEmailAlertNoFill.php:91]: Nice catch!

**Review by anarute — APPROVED** [review]
Looks good to me, thanks for the clean up

File: jobs/sendEmailAlertNoFill.php (hunk 3 above)
Comment (jaragunde) [reply jobs/sendEmailAlertNoFill.php:91]: I failed to catch it when I reviewed that code back in the day :sweat_smile:

**Pull request merged**